### PR TITLE
Adding dartdoc to flame_audio

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adding dartdoc to flame_audio.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame_audio/example/lib/main.dart [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// ignore_for_file: public_member_api_docs
```
Comment (luanpotter) [inline packages/flame_audio/example/lib/main.dart:1]: can't we override the rule on the example project analysis options? I think it's bad to have to ignore a rule on every example file. the rule just doesn't apply to the project

File: packages/flame_audio/lib/audio_pool.dart [hunk 2]
```diff
@@ -14,8 +15,13 @@ class AudioPool {
   final Map<String, AudioPlayer> _currentPlayers = {};
   final List<AudioPlayer> _availablePlayers = [];
 
+  /// The path of the sound of this pool
   final String sound;
+
+  /// If the poop is repeating
```
Comment (spydon) [inline packages/flame_audio/lib/audio_pool.dart:21]: Hahah, I hate when the poop is repeating!
```suggestion
  /// If the pool is repeating
```

**Review by spydon — APPROVED** [review]
Same here, we should settle for either ending all dartdocs with period or not, otherwise it looks good! :)

**Comment by st-pasha** [comment]
> Same here, we should settle for either ending all dartdocs with period or not, otherwise it looks good! :)

Flutter style guide recommends using periods: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar

File: packages/flame_audio/lib/audio_pool.dart [hunk 3]
```diff
@@ -52,6 +59,7 @@ class AudioPool {
     return instance;
   }
 
+  /// Starts playing the audio, returns a function that can stopped the audio.
```
Comment (luanpotter) [inline packages/flame_audio/lib/audio_pool.dart:62]:
```suggestion
  /// Starts playing the audio, returns a function that can stop the audio.
```

File: packages/flame_audio/lib/bgm.dart [hunk 4]
```diff
@@ -3,19 +3,28 @@ import 'dart:io';
 import 'package:audioplayers/audioplayers.dart';
 import 'package:flutter/widgets.dart';
 
+/// {@template _bgm}
```
Comment (luanpotter) [inline packages/flame_audio/lib/bgm.dart:6]: just curious, what is this?

File: packages/flame_audio/lib/bgm.dart [hunk 5]
```diff
@@ -3,19 +3,28 @@ import 'dart:io';
...
 class Bgm extends WidgetsBindingObserver {
   bool _isRegistered = false;
+
+  /// [AudioCache] instance of the [Bgm].
   late AudioCache audioCache;
+
+  /// The [AudioPlayer] instance that is currently playing the audio.
   AudioPlayer? audioPlayer;
+
+  /// If the [Bgm] is playing or not.
```
Comment (luanpotter) [inline packages/flame_audio/lib/bgm.dart:24]: my most misspelled word :P
```suggestion
  /// Whether [Bgm] is playing or not.
```

File: packages/flame_audio/lib/bgm.dart [hunk 6]
```diff
@@ -71,7 +80,7 @@ class Bgm extends WidgetsBindingObserver {
     }
   }
 
-  /// Pauses the background music without unloading or resetting the audio
+  /// Pauses the background music without unloading or resetting the audio.
```
Comment (luanpotter) [inline packages/flame_audio/lib/bgm.dart:83]: I don't think this is right

File: packages/flame_audio/lib/flame_audio.dart [hunk 7]
```diff
@@ -15,7 +15,7 @@ class FlameAudio {
     return audioCache.play(file, volume: volume, mode: PlayerMode.LOW_LATENCY);
   }
 
-  /// Plays, and keep looping the given [file]
+  /// Plays, and keep looping the given [file].
```
Comment (luanpotter) [inline packages/flame_audio/lib/flame_audio.dart:18]:
```suggestion
  /// Plays, and keeps looping the given [file].
```

File: packages/flame_audio/lib/flame_audio.dart [hunk 8]
```diff
@@ -31,7 +31,7 @@ class FlameAudio {
   ///
   /// NOTE: Length audio files on Android have an audio gap between loop
   /// iterations, this happens due to limitations on Android's native media
-  /// player features, if you need a gapless loop, prefer the loop method
+  /// player features, if you need a gapless loop, prefer the loop method.
```
Comment (luanpotter) [inline packages/flame_audio/lib/flame_audio.dart:34]:
```suggestion
  /// player features. If you need a gapless loop, prefer the loop method.
```

File: packages/flame_audio/lib/bgm.dart (hunk 4 above)
Comment (erickzanardo) [reply packages/flame_audio/lib/bgm.dart:6]: It allows for docs to be reused, if you look below, it is used on the constructor with `@macro _bgm`

File: packages/flame_audio/lib/bgm.dart (hunk 6 above)
Comment (erickzanardo) [reply packages/flame_audio/lib/bgm.dart:83]: Just added the period for consistency

Comment (luanpotter) [reply packages/flame_audio/lib/bgm.dart:83]: it's breaking the sentence though

**Review by luanpotter — APPROVED** [review]
LGTM with one final comment

File: packages/flame_audio/lib/bgm.dart (hunk 6 above)
Comment (erickzanardo) [inline packages/flame_audio/lib/bgm.dart:83]:
```suggestion
  /// Pauses the background music without unloading or resetting the audio
```

**Pull request merged**